### PR TITLE
Make features explicit, error fixes,

### PR DESCRIPTION
**Pull request by nyurik**
* Use `thiserror` for all errors to allow better error reporting
* Make all HTTP errors into proper objects without string formatting -- formatting is delayed until printing, or if not logged by end user, works much faster
* add justfile to simplify development (optional, but useful)
* cleanup CI to make it far simpler (that GH action is not maintained, and not worth it)
* Made cargo.toml explicitly declare all features - instead of auto-exposing optional dependencies as features
* Lastly, it seems `fmmap` has not been updated for a long time, and because of that it no longer compiles in some cases because of the dependency problems.  I submitted a PR, hopefully will be merged/published soonish.

### CODE REVIEW

**Comment by nyurik** [comment]
P.S. My fmmap PR was merged (impressive speed!), but i'm still a bit concerned - that repo is not often maintained, whereas https://github.com/RazrFalcon/memmap2-rs seems to be the latest "standard" in the memmap crates.  Is there a reason for fmmap?

**Comment by lseelenbinder** [comment]
The simple answer about the mmap library is async support @nyurik. It doesn't look like memmap2 supports async, and I didn't want to litter the library with block_in_place calls.

File: src/error.rs [hunk 1]
```diff
@@ -1,28 +1,53 @@
 use std::string::FromUtf8Error;
 
-#[derive(Debug)]
+// use std::string::FromUtf8Error;
```
Comment (lseelenbinder) [inline src/error.rs:3]: Remove?

**Comment by lseelenbinder** [comment]
Looks good once tests pass. 👍

**Review by lseelenbinder — APPROVED** [review]

**Comment by nyurik** [comment]
I'll wait for a day or two to see if fmmap is released - feels kinda bad to use an unreleased dependency

**Pull request merged**